---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

```{r, echo = FALSE, fig.align='center'}
knitr::include_graphics("man/figures/microshades_logo.png")
```


<!-- badges: start -->
<!-- badges: end -->



The goal of microshades is to provide a custom color organization palette that can be applied to enhance current data visualization methods. 
To increase accessibility of visuals created, users may select to use color vision deficient (CVD) friendly colors.


This package includes functions that work with phyloseq organized microbiome data to create stacked bar plots with advanced color organization.  With microshades color organization, plots are more intuitive and accessible.


Additionally, microshades colors can be incorporated into any plot by using our palette selecting functions.


## Installation

``` r
remotes::install_github("KarstensLab/microshades")
```

## The shades

Here is a list of the microshades palettes, including the CVD friendly shades

```{r shades, warning = FALSE}
library(microshades)

# main shades
names(microshades_palettes)

# cvd shades
names(microshades_cvd_palettes)

```

To view any shade, use the function `microshades_palette` and specify the shade to view.

```{r blue, fig.height = 1, fig.width = 4}
microshades_palette("micro_blue")
```

The CVD shades use the same function.

```{r cvd_blue, fig.height = 1, fig.width = 4}
microshades_palette("micro_cvd_blue")
```


## Phyloseq useage

For detailed tutorials on how to use microshades function with phyloseq objects, please reference vignette articles on the [website](https://KarstensLab.github.io/microshades).


Here is an example of the plot generated with microshades on Curated Metagenomic Data of the Human Microbiome.

```{r, echo = FALSE, fig.align='center'}
knitr::include_graphics("man/figures/plot_CMD.png")
```

microshades uses coloring to correspond with taxonomic group and subgroup levels. 
In this example, the phylum and genus information are explored. 
Darker shades indicate the most abundant genera for each phylum, and lighter shades are less abundant.
Users can additionally reorder the samples based on a specified taxonomic rank and name, in this example by genus "Eubacterium" abundance.


## Apply the microshades palette to other plots

To only use and apply a microshades palette color to an existing graph, use `scale_fill_manual()`.
Below is an example of how to apply the color palettes to a different type of plot.

```{r plot example}
library(palmerpenguins)
library(dplyr)
library("ggplot2")
 
data(package = 'palmerpenguins')

ggplot(penguins, aes(species, fill = island)) + geom_bar() + 
         scale_fill_manual(values = microshades_palette("micro_green"))
       
```


```{r}
penguins_mod <- penguins %>% mutate(combination_variable = paste(species, year, sep = "-"))

hex_values <-c(microshades_palette("micro_green", 3, lightest = FALSE), 
               microshades_palette("micro_blue", 3, lightest = FALSE), 
               microshades_palette("micro_purple", 3, lightest = FALSE))

ggplot(penguins_mod, aes(x = flipper_length_mm,
                            y = body_mass_g)) +
  geom_point(aes(color = combination_variable)) +
  theme_minimal() +
  scale_color_manual(values = hex_values, na.translate = FALSE) +
  labs(title = "Penguin flipper and body mass",
       x = "Flipper length (mm)",
       y = "Body mass (g)",
       color = "Penguin Species and Year") +
  theme(legend.position = "bottom",
        legend.background = element_rect(fill = "white", color = NA),
        plot.title.position = "plot",
        plot.caption = element_text(hjust = 0, face= "italic"),
        plot.caption.position = "plot") +
  facet_wrap(~species)

```

```{r}


  
bill_len_dep <- ggplot(data = penguins_mod,
                         aes(x = bill_length_mm,
                             y = bill_depth_mm,
                             group = combination_variable)) +
  geom_point(aes(color = combination_variable),
             size = 3,
             alpha = 0.8) +
  geom_smooth(method = "lm", se = FALSE, aes(color = combination_variable)) +
  theme_minimal() +
  scale_color_manual(values = hex_values, na.translate = FALSE) +
  labs(title = "Penguin bill dimensions",
       subtitle = "Bill length and depth for Adelie, Chinstrap and Gentoo Penguins at Palmer Station LTER over the years 2007-2009",
       x = "Bill length (mm)",
       y = "Bill depth (mm)",
       color = "Penguin species",
       shape = "Penguin species") +
  theme(legend.position = "right",
        legend.background = element_rect(fill = "white", color = NA),
        plot.title.position = "plot",
        plot.caption = element_text(hjust = 0, face= "italic"),
        plot.caption.position = "plot")

bill_len_dep

```


