<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Human Microbiome Project • microshades</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Human Microbiome Project">
<meta property="og:description" content="microshades">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-196876408-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-196876408-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">microshades</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/microshades-GP.html">Global Patterns Data</a>
    </li>
    <li>
      <a href="../articles/microshades-HMP.html">Human Microbiome Project</a>
    </li>
    <li>
      <a href="../articles/microshades-HMP2.html">Human Microbiome Project 2</a>
    </li>
    <li>
      <a href="../articles/non-microbiome_data.html">Palmer Penguins Data Example</a>
    </li>
    <li>
      <a href="../articles/titanic_data_example.html">Titanic Data Example</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/KarstensLab/microshades/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Human Microbiome Project</h1>
                        <h4 data-toc-skip class="author">Erin Dahl</h4>
            
            <h4 data-toc-skip class="date">October 31, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/KarstensLab/microshades/blob/HEAD/vignettes/microshades-HMP.Rmd" class="external-link"><code>vignettes/microshades-HMP.Rmd</code></a></small>
      <div class="hidden name"><code>microshades-HMP.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="human-microbiome-project-16s-data-vignette">Human Microbiome Project 16S Data Vignette<a class="anchor" aria-label="anchor" href="#human-microbiome-project-16s-data-vignette"></a>
</h2>
<p>This vignette uses 16S rRNA sequencing data from variable region
v1-v3 in the Human Microbiome Project. Stool samples from male subjects
will be explored at the Phylum &amp; Family taxonomic levels.</p>
<p>This data is available in the HMP16SData library. To download use
<code>BiocManager::install("HMP16SData")</code>.</p>
<p>Learn more about the HMP16S Data <a href="https://bioconductor.org/packages/release/data/experiment/html/HMP16SData.html" class="external-link">here</a></p>
<p>Additionally, the package <a href="https://github.com/mikemc/speedyseq" class="external-link">speedyseq</a> is necessary to
use the function <code><a href="../reference/prep_mdf.html">prep_mdf()</a></code>. The package speedyseq provides
faster versions of phyloseq’s plotting and taxonomic merging functions.
Alternatively, the phyloseq object can be melted and transformed by
using phyloseq functions <code><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom()</a></code> and/or
<code><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts()</a></code>, and melted by using
<code><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt()</a></code>.</p>
<div class="section level3">
<h3 id="load-the-necessary-packages-for-this-tutorial">Load the necessary packages for this tutorial<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages-for-this-tutorial"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/HMP16SData" class="external-link">HMP16SData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://karstenslab.github.io/microshades" class="external-link">microshades</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-and-subset-the-hmp-data">Load and subset the HMP data<a class="anchor" aria-label="anchor" href="#load-and-subset-the-hmp-data"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">v1v3_stool</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/HMP16SData/man/V13.html" class="external-link">V13</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span>select <span class="op">=</span> <span class="va">HMP_BODY_SUBSITE</span> <span class="op">==</span> <span class="st">"Stool"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span>select <span class="op">=</span> <span class="va">SEX</span> <span class="op">==</span> <span class="st">"Male"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## snapshotDate(): 2020-10-27</span></code></pre>
<pre><code><span class="co">## see ?HMP16SData and browseVignettes('HMP16SData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps_v1v3_stool</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/HMP16SData/man/as_phyloseq.html" class="external-link">as_phyloseq</a></span><span class="op">(</span><span class="va">v1v3_stool</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="apply-microshades-functions">Apply microshades functions<a class="anchor" aria-label="anchor" href="#apply-microshades-functions"></a>
</h3>
<p>Begin using microshades functions to evaluate abundance and apply
advanced color organization at the Phylum Family level.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use microshades function prep_mdf to agglomerate, normalize, and melt the phyloseq object</span>
<span class="va">mdf_prep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_mdf.html">prep_mdf</a></span><span class="op">(</span><span class="va">ps_v1v3_stool</span>, subgroup_level <span class="op">=</span> <span class="st">"FAMILY"</span><span class="op">)</span>

<span class="co"># Create a color object for the specified data</span>
<span class="va">color_obj_v1v3_stool</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_color_dfs.html">create_color_dfs</a></span><span class="op">(</span><span class="va">mdf_prep</span>, group_level <span class="op">=</span> <span class="st">"PHYLUM"</span>, subgroup_level <span class="op">=</span> <span class="st">"FAMILY"</span>, cvd <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>


<span class="co"># Extract</span>
<span class="va">mdf_v1v3_stool</span> <span class="op">&lt;-</span> <span class="va">color_obj_v1v3_stool</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_v1v3_stool</span> <span class="op">&lt;-</span> <span class="va">color_obj_v1v3_stool</span><span class="op">$</span><span class="va">cdf</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot">Plot<a class="anchor" aria-label="anchor" href="#plot"></a>
</h3>
<p>The dataframe <code>mdf_v1v3_stool</code> contains sample data and
abundance info. The dataframe <code>cdf_v1v3_stool</code> stores the
color mapping information used for plotting.</p>
<p><code>plot_microshades</code> produces a stacked barplot with ordered
subgroup taxonomy. The darkest shade is the most abundant, and the
lightest shade is the least abundant. (excluding the “other” subgroup
from each of the selected groups)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_v1v3_stool</span>, <span class="va">cdf_v1v3_stool</span>, group_label <span class="op">=</span> <span class="st">"Phylum Family"</span><span class="op">)</span>

<span class="va">plot_1</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="microshades-HMP_files/figure-html/unnamed-chunk-4-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="reorder-samplesgroup-levels">Reorder Samples/Group levels<a class="anchor" aria-label="anchor" href="#reorder-samplesgroup-levels"></a>
</h3>
<p>The plot above indicates that Bacteroidaceae is the most abundant
family in the Bacteriodetes phylum in this dataset. To see samples
arranged in order of Bacteroidaceae abundance (or any genus listed in
the legend above) use the <code><a href="../reference/reorder_samples_by.html">reorder_samples_by()</a></code> function and
then plot. This function can also reorder the Group levels based on
overall group level abundance, if desired steeing the
sink_abundant_groups to true.</p>
<p>Reodering the samples can help users notice additional visual
trends.</p>
<div class="section level4">
<h4 id="reorder-the-samples-by-bacteroidaceae-abundance">Reorder the samples by Bacteroidaceae Abundance<a class="anchor" aria-label="anchor" href="#reorder-the-samples-by-bacteroidaceae-abundance"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reorder_samples_by will change the order of samples based on an abundance of a specified subgroup taxonomy</span>
<span class="va">new_sample_order_v1v3_stool</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reorder_samples_by.html">reorder_samples_by</a></span><span class="op">(</span><span class="va">mdf_v1v3_stool</span>, <span class="va">cdf_v1v3_stool</span>, order <span class="op">=</span> <span class="st">"Bacteroidaceae"</span>, group_level <span class="op">=</span> <span class="st">"PHYLUM"</span>, subgroup_level <span class="op">=</span> <span class="st">"FAMILY"</span>, sink_abundant_groups <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">mdf_new_smample_order</span> <span class="op">&lt;-</span><span class="va">new_sample_order_v1v3_stool</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_new_smample_order</span> <span class="op">&lt;-</span><span class="va">new_sample_order_v1v3_stool</span><span class="op">$</span><span class="va">cdf</span>

<span class="va">plot_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_new_smample_order</span>, <span class="va">cdf_new_smample_order</span>, group_label <span class="op">=</span> <span class="st">"Phylum Family"</span><span class="op">)</span>

<span class="va">plot_2</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-HMP_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
</div>
<div class="section level4">
<h4 id="reorder-by-abundance-of-group-levels-only">Reorder by Abundance of Group levels Only<a class="anchor" aria-label="anchor" href="#reorder-by-abundance-of-group-levels-only"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reorder_samples_by will change the order of samples based on an abundance of a specified subgroup taxonomy</span>
<span class="co"># The default subgroup_level is "Genus" </span>
<span class="va">new_group_order_v1v3_stool</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reorder_samples_by.html">reorder_samples_by</a></span><span class="op">(</span><span class="va">mdf_v1v3_stool</span>, <span class="va">cdf_v1v3_stool</span>, group_level <span class="op">=</span> <span class="st">"PHYLUM"</span>, subgroup_level <span class="op">=</span> <span class="st">"FAMILY"</span>, sink_abundant_groups <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">mdf_new_group_order</span> <span class="op">&lt;-</span><span class="va">new_group_order_v1v3_stool</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_new_group_order</span> <span class="op">&lt;-</span><span class="va">new_group_order_v1v3_stool</span><span class="op">$</span><span class="va">cdf</span>

<span class="va">plot_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_new_group_order</span>, <span class="va">cdf_new_group_order</span>, group_label <span class="op">=</span> <span class="st">"Phylum Family"</span><span class="op">)</span>

<span class="va">plot_3</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-HMP_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
</div>
<div class="section level4">
<h4 id="reorder-by-both-samples-and-abundance-of-group-levels">Reorder by both Samples and Abundance of Group levels<a class="anchor" aria-label="anchor" href="#reorder-by-both-samples-and-abundance-of-group-levels"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reorder_samples_by will change the order of samples based on Family an abundance of "Bacteroidaceae" and reorder the Phylum to sink the most abundant groups</span>
<span class="va">reordered_v1v3_stool</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reorder_samples_by.html">reorder_samples_by</a></span><span class="op">(</span><span class="va">mdf_v1v3_stool</span>, <span class="va">cdf_v1v3_stool</span>, order <span class="op">=</span> <span class="st">"Bacteroidaceae"</span>, group_level <span class="op">=</span> <span class="st">"PHYLUM"</span>, subgroup_level <span class="op">=</span> <span class="st">"FAMILY"</span>, sink_abundant_groups <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">mdf_reordered</span> <span class="op">&lt;-</span><span class="va">reordered_v1v3_stool</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_reordered</span> <span class="op">&lt;-</span><span class="va">reordered_v1v3_stool</span><span class="op">$</span><span class="va">cdf</span>

<span class="va">plot_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_reordered</span>, <span class="va">cdf_reordered</span>, group_label <span class="op">=</span> <span class="st">"Phylum Family"</span><span class="op">)</span>

<span class="va">plot_3</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-HMP_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Erin Dahl, Emory Neer, Lisa Karstens.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
