<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>microshades-CMD â€¢ microshades</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="microshades-CMD">
<meta property="og:description" content="microshades">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">microshades</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/microshades-CMD.html">microshades-CMD</a>
    </li>
    <li>
      <a href="../articles/microshades-GP.html">microshades-GP</a>
    </li>
    <li>
      <a href="../articles/microshades-HMP.html">microshades-HMP</a>
    </li>
    <li>
      <a href="../articles/microshades-HMP2.html">microshades-HMP2</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="microshades-CMD_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="microshades-CMD_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>microshades-CMD</h1>
                        <h4 class="author">Erin Dahl</h4>
            
            <h4 class="date">March 19, 2021</h4>
      
      
      <div class="hidden name"><code>microshades-CMD.Rmd</code></div>

    </div>

    
    
<div id="curated-metagenomic-data-vignette" class="section level1">
<h1 class="hasAnchor">
<a href="#curated-metagenomic-data-vignette" class="anchor"></a>Curated Metagenomic Data Vignette</h1>
<p>This vignette uses Curated Metagenomic Data of the Human Microbiome.</p>
<p>This data is available in the curatedMetagenomicData library. To download use <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html">BiocManager::install("curatedMetagenomicData")</a></code>.</p>
<p>Learn more about the Curated Metagenomic Data <a href="https://github.com/waldronlab/curatedMetagenomicData">here</a>. Secondary link <a href="https://bioconductor.org/packages/release/data/experiment/html/curatedMetagenomicData.html" class="uri">https://bioconductor.org/packages/release/data/experiment/html/curatedMetagenomicData.html</a></p>
<div id="load-the-necessary-packages-for-this-tutorial" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-necessary-packages-for-this-tutorial" class="anchor"></a>Load the necessary packages for this tutorial</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microshades</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-datasets-and-convert-to-phyloseq-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#load-datasets-and-convert-to-phyloseq-objects" class="anchor"></a>Load datasets and convert to phyloseq objects</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load the CMD data </span>
<span class="va">britol</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/BritoIL_2016.html">BritoIL_2016.metaphlan_bugs_list.stool</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ps_britol</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq</a></span> <span class="op">(</span><span class="va">britol</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: `data_frame()` is deprecated as of tibble 1.1.0.
## Please use `tibble()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">HMP</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/HMP_2012.html">HMP_2012.metaphlan_bugs_list.stool</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ps_HMP</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq</a></span> <span class="op">(</span><span class="va">HMP</span><span class="op">)</span></code></pre></div>
<p>The <code>ps_britol</code> object contains 112 samples, and <code>ps_HMP</code> contains 141 samples. To focus on a smaller sample size, use phyloseq function <code><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples()</a></code>. In this scenario, only schoolage subjects samples will be included.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset the samples</span>
<span class="va">ps_britol_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_britol</span>,<span class="va">age_category</span> <span class="op">==</span> <span class="st">"schoolage"</span><span class="op">)</span>

<span class="va">ps_HMP_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_HMP</span>,<span class="va">age_category</span> <span class="op">==</span> <span class="st">"schoolage"</span><span class="op">)</span></code></pre></div>
<p>Now <code>ps_britol_sub</code> object contains 18 samples, and <code>ps_HMP_sub</code> contains 6 samples. Begin using microshades functions to evaluate abundance and apply advanced color organization.</p>
</div>
<div id="apply-the-microshades-functions-to-the-britol-dataset-" class="section level2">
<h2 class="hasAnchor">
<a href="#apply-the-microshades-functions-to-the-britol-dataset-" class="anchor"></a>Apply the microshades functions to the Britol dataset.</h2>
<p>To orient the shades from the top darkest to lightest instead of from the bottom darkest to lightest, use the <code>top_orientation = TRUE</code> parameter in the <code><a href="../reference/create_color_dfs.html">create_color_dfs()</a></code></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use microshades function prep_mdf to agglomerate, normalize, and melt the phyloseq object</span>
<span class="va">mdf_britol_pre</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_mdf.html">prep_mdf</a></span><span class="op">(</span><span class="va">ps_britol_sub</span><span class="op">)</span>

<span class="co"># Create a color object for the specified data</span>
<span class="va">color_objs_britol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_color_dfs.html">create_color_dfs</a></span><span class="op">(</span><span class="va">mdf_britol_pre</span>, top_orientation <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Extract</span>
<span class="va">mdf_britol</span><span class="op">&lt;-</span> <span class="va">color_objs_britol</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_britol</span> <span class="op">&lt;-</span> <span class="va">color_objs_britol</span><span class="op">$</span><span class="va">cdf</span></code></pre></div>
<div id="plot-britol" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-britol" class="anchor"></a>Plot Britol</h3>
<p>The dataframe <code>mdf_britol</code> contains sample data and abundance info. The dataframe <code>cdf_britol</code> stores the color mapping information used for plotting.</p>
<p><code>plot_microshades</code> produces a stacked barplot with ordered subgroup taxonomy. The darkest shade is the most abundant, and the lightest shade is the least abundant. (excluding the "other" subgroup from each of the selected groups)</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_britol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_britol</span>, <span class="va">cdf_britol</span><span class="op">)</span>

<span class="va">plot_britol</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-5-1.png" width="576"></p>
</div>
</div>
<div id="reassign-colors-as-desired" class="section level2">
<h2 class="hasAnchor">
<a href="#reassign-colors-as-desired" class="anchor"></a>Reassign colors as desired</h2>
<p>To change the colors assignment, use the function <code>color_reassign</code> to specify the groups and colors in order of reassignment. For example, we can change the color assignment of Bacteroidetes to <code>micro_green</code> and the color assignment of Proteobacteria to <code>micro_blue</code></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">new_cdf_britol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/color_reassign.html">color_reassign</a></span><span class="op">(</span><span class="va">cdf_britol</span>, 
                                 group_assignment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bacteroidetes"</span>, <span class="st">"Proteobacteria"</span><span class="op">)</span>,
                                 color_assignment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"micro_green"</span>, <span class="st">"micro_blue"</span><span class="op">)</span><span class="op">)</span>

<span class="va">new_plot_britol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_britol</span>, <span class="va">new_cdf_britol</span><span class="op">)</span>

<span class="va">new_plot_britol</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="apply-the-microshades-functions-to-the-hmp-dataset-" class="section level2">
<h2 class="hasAnchor">
<a href="#apply-the-microshades-functions-to-the-hmp-dataset-" class="anchor"></a>Apply the microshades functions to the HMP dataset.</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use microshades function prep_mdf to agglomerate, normalize, and melt the phyloseq object</span>
<span class="va">mdf_HMP_pre</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_mdf.html">prep_mdf</a></span><span class="op">(</span><span class="va">ps_HMP_sub</span><span class="op">)</span>

<span class="co"># Create a color object for the specified data</span>
<span class="va">color_objs_HMP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_color_dfs.html">create_color_dfs</a></span><span class="op">(</span><span class="va">mdf_HMP_pre</span><span class="op">)</span>

<span class="co"># Extract</span>
<span class="va">mdf_HMP</span> <span class="op">&lt;-</span> <span class="va">color_objs_HMP</span><span class="op">$</span><span class="va">mdf</span>
<span class="va">cdf_HMP</span> <span class="op">&lt;-</span> <span class="va">color_objs_HMP</span><span class="op">$</span><span class="va">cdf</span></code></pre></div>
<div id="plot-hmp" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-hmp" class="anchor"></a>Plot HMP</h3>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_HMP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_HMP</span>, <span class="va">cdf_HMP</span><span class="op">)</span>

<span class="va">plot_HMP</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
<p>Examine Legends</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">HMP_legend</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/get_legend.html">get_legend</a></span><span class="op">(</span><span class="va">plot_HMP</span><span class="op">)</span>

<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">HMP_legend</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">britol_legend</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/get_legend.html">get_legend</a></span><span class="op">(</span><span class="va">new_plot_britol</span><span class="op">)</span>

<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">britol_legend</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<p>Each legend is configured to reflect the dataset of the plot. Notice that the Britol and HMP Legends are similar, but not the exact same, due to different top abundances for genera in each phylum.</p>
</div>
</div>
<div id="use-britol-legend-for-hmp-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#use-britol-legend-for-hmp-plot" class="anchor"></a>Use Britol legend for HMP Plot</h2>
<p>If the plots are being compared, it may be confusing visually if the genus shading colors account for different genera between the two plots. In this case, the function <code><a href="../reference/match_cdf.html">match_cdf()</a></code> can apply the color factoring information from one processed mdf to a different, unprocessed mdf.</p>
<p>To use the Britol color legend with the HMP data, apply the Britol color factoring to the HMP data.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mdf_HMP_matched</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/match_cdf.html">match_cdf</a></span><span class="op">(</span><span class="va">mdf_HMP_pre</span>, <span class="va">mdf_britol</span><span class="op">)</span>

<span class="co"># Use the Britol cdf for color reference</span>
<span class="va">plot_HMP_matched</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_microshades.html">plot_microshades</a></span><span class="op">(</span><span class="va">mdf_HMP_matched</span>, <span class="va">new_cdf_britol</span><span class="op">)</span>

<span class="va">plot_HMP_matched</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span>, expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="st">"cm"</span><span class="op">)</span>, text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="microshades-CMD_files/figure-html/unnamed-chunk-10-1.png" width="360"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Erin Dahl, Lisa Karstens, Eric Leung, Emory Neer, Kate Bowie.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
